packages/nuxt/src/app/utils.ts

```diff
@@ -1,4 +1,49 @@
-/** @since 3.9.0 */
+import { useNuxtApp } from './nuxt'
+
 export function toArray<T> (value: T | T[]): T[] {
   return Array.isArray(value) ? value : [value]
 }
+
+type CallbackFn = () => void
+type ObserveFn = (element: Element, callback: CallbackFn) => () => void
+
+export function useObserver (): { observe: ObserveFn } | undefined {
+  if (import.meta.server) { return }
```

Comment:
Having a single return type would be better, we could make a noop `observe` function, otherwise, end users need to do type juggling

Reply:
Yea I was thinking about that, could that cause any issues if the server render returns a blank observer function?

Reply:
Should I create another overload of this that won't cache the observer in nuxtApp._observer? To allow different options?

Reply:
Reused the same one, saw that the logic matches